pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -2306,6 +2306,310 @@ def init_arc_param(arctype):
+        * model.model_units["time"],
+        doc="Total deliveries to completions pads to meet completions demand [volume]",
+    )
+
+    model.e_CompletionsSourcedFrac = Expression(
+        expr=model.v_F_TotalSourced / model.e_TotalCompletionsDeliveries,
+        doc="Fraction of completions deliveries using externally sourced water [fraction]",
+    )
+
+    model.e_CompletionsReusedFrac = Expression(
+        expr=model.v_F_TotalReused / model.e_TotalCompletionsDeliveries,
+        doc="Fraction of completions deliveries using reused water [fraction]",
+    )
+
+    model.e_TotalResidualWater = Expression(
+        expr=sum(
+            sum(model.v_F_ResidualWater[r, t] for t in model.s_T) for r in model.s_R
+        )
+        * model.model_units["time"],
+        doc="Total residual water from treatment sites [volume]",
+    )
+
+    model.e_TotalResidualDesalinationWater = Expression(
+        expr=sum(
+            sum(model.v_F_ResidualWater[r, t] for t in model.s_T)
+            for r in model.s_R
+            if model.p_chi_DesalinationSites[r] == 1
+        )
+        * model.model_units["time"],
+        doc="Total residual water from desalination treatment sites [volume]",
+    )
+
+    model.e_TotalResidualNonDesalWater = Expression(
+        expr=sum(
+            sum(model.v_F_ResidualWater[r, t] for t in model.s_T)
+            for r in model.s_R
+            if model.p_chi_DesalinationSites[r] == 0
+        )
+        * model.model_units["time"],
+        doc="Total residual water from non-desalination treatment sites [volume]",
+    )
+
+    model.e_TotalReused = Expression(
+        expr=sum(
+            sum(model.v_F_ReuseDestination[p, t] for p in model.s_CP) for t in model.s_T
+        )
+        * model.model_units["time"],
+        doc="Total water reused to meet completions demand [volume]",
+    )
+
+    model.e_TotalReusedInternal = Expression(
+        expr=sum(
+            sum(
+                model.v_F_ReuseDestination[p, t]
+                for p in model.s_CP
+                if model.p_chi_OutsideCompletionsPad[p] == 0
+            )
+            for t in model.s_T
+        )
+        * model.model_units["time"],
+        doc="Total water reused to meet completions demand internal to system [volume]",
+    )
+
+    model.e_TotalReusedExternal = Expression(
+        expr=sum(
+            sum(
+                model.v_F_ReuseDestination[p, t]
+                for p in model.s_CP
+                if model.p_chi_OutsideCompletionsPad[p] == 1
+            )
+            for t in model.s_T
+        )
+        * model.model_units["time"],
+        doc="Total water reused to meet completions demand external to system [volume]",
+    )
+
+    model.e_TotalEvaporated = Expression(
+        expr=sum(
+            sum(model.v_F_StorageEvaporationStream[s, t] for s in model.s_S)
+            for t in model.s_T
+        )
+        * model.model_units["time"],
+        doc="Total water evaporated [volume]",
+    )
```

Comment:
Here we assume that the storage facility is open and exposed to water?

Reply:
Yes, that's the assumption. The way we have implemented evaporation is something we should revisit in the future, however - we have issue #158 open for that purpose.